lib/AArch64Base.ml

```diff
@@ -418,27 +418,57 @@ let barrier_compare = compare
 module IC = struct
   type funct = I
   let pp_funct = function I -> "I"
+  let fold_funct f k = f I k
 
   type typ = ALL | VA
   let pp_typ = function | ALL -> "ALL" | VA -> "VA"
+  let fold_typ f k =
```

Comment:
```suggestion
  let fold_typ f k =
    k
    |> f ALL
    |> f VA
```
As a completely optional point: this computation could be a nice opportunity to use piping (|>). This might work for many of the new `fold_*` functions. That said, the readability/aesthetics gains with that are rather subjective; please do ignore it if it looks like encryption to you!

Reply:
That's a very good point @artkhyzha, I'll change all fold_* functions to use |>. Thanks!